libsolidity/formal/CHC.cpp

```diff
@@ -907,18 +917,20 @@ void CHC::defineInterfacesAndSummaries(SourceUnit const& _source)
 					auto state1 = stateVariablesAtIndex(1, *contract);
 					auto state2 = stateVariablesAtIndex(2, *contract);
 
+					auto errorPre = errorFlag().currentValue();
 					auto nondetPre = smt::nondetInterface(iface, *contract, m_context, 0, 1);
+					auto errorPost = errorFlag().increaseIndex();
 					auto nondetPost = smt::nondetInterface(iface, *contract, m_context, 0, 2);
 
-					vector<smtutil::Expression> args{errorFlag().currentValue(), state().thisAddress(), state().abi(), state().crypto(), state().tx(), state().state(1)};
+					vector<smtutil::Expression> args{errorPost, state().thisAddress(), state().abi(), state().crypto(), state().tx(), state().state(1)};
 					args += state1 +
 						applyMap(function->parameters(), [this](auto _var) { return valueAtIndex(*_var, 0); }) +
 						vector<smtutil::Expression>{state().state(2)} +
 						state2 +
 						applyMap(function->parameters(), [this](auto _var) { return valueAtIndex(*_var, 1); }) +
 						applyMap(function->returnParameters(), [this](auto _var) { return valueAtIndex(*_var, 1); });
 
-					connectBlocks(nondetPre, nondetPost, (*m_summaries.at(contract).at(function))(args));
+					connectBlocks(nondetPre, nondetPost, errorPre == 0 && (*m_summaries.at(contract).at(function))(args));
```

Comment:
To confirm, this rule then looks like
`nondet_interface(0, abi, crypto, stateVars, stateVars') && f_summary(error, abi, crypto, stateVars', stateVars'') => nondet_interface(error, abi, crypto, stateVars, stateVars''`
correct?

Reply:
To be precise, it looks like `nondet_interface(error, abi, crypto, stateVars, stateVars') && error == 0 && f_summary(error', abi, crypto, stateVars', stateVars'') => nondet_interface(error', abi, crypto, stateVars, stateVars''`

Reply:
Ok, yes. I wanted to be sure that the function error was the one being propagated. Looks good.